pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/util/SegmentDeletionManagerTest.java

```diff
@@ -629,26 +629,34 @@ public boolean mkdir(URI uri)
         throws IOException {
       // create a new table Dir if the path ends with /
       if (uri.getPath().endsWith("/")) {
-        _tableDirs.put(uri.getPath(), new HashSet<>());
+        synchronized (_tableDirs) {
```

Comment:
What's the threading model here? Can you add a comment on why this synchronized block is needed? And it's needed even though this `FakePinotFs` is only being used for `testRemoveDeletedSegmentsForGcsPinotFS` which is a single threaded test?

Reply:
Is it because of contention with the `SegmentDeletionManager`'s single threaded executor which does some background deletes?

Reply:
SegmentDeletionManager uses a single-threaded ScheduledExecutorService for async deletion. In the test we call removeAgedDeletedSegments(), which submits deletion work to that executor while the test thread concurrently calls PinotFS.exists()/listFiles(). FakePinotFs synchronizes access to _tableDirs to avoid races/CMEs and ensure visibility between the executor thread and the test thread.